src/actions/postsActions.js

```diff
@@ -86,12 +76,14 @@ export function getPosts(): ThunkAction {
 export function updatePost(post: Post): ThunkAction {
   return dispatch => {
     retrieveData("data", async (err, postsData) => {
-      const posts = JSON.parse(postsData);
-      const index = getIndex(posts, post.id);
-      if (index !== -1) {
-        posts[index]["description"] = post.description;
-        posts[index]["tag"] = post.tag;
-      }
+      const InitialPosts = JSON.parse(postsData);
+      const posts = InitialPosts.map(obj => {
+        if (obj.id === post.id) {
```

Comment:
Don't change initial argument. Clone object with spread operator, then change it